analyzer/emerald/incoming.go

```diff
@@ -348,7 +352,7 @@ func extractRound(sigContext signature.Context, b *block.Block, txrs []*sdkClien
 	return &blockData, nil
 }
 
-func emitRoundBatch(batch *storage.QueryBatch, round int64, blockData *BlockData) {
+func emitRoundBatch(batch *storage.QueryBatch, round uint64, blockData *BlockData) {
```

Comment:
Apologies if this was asked already, but what's the reason for switching to `uint64` here? 

Intuitively it makes sense since the round can't be negative. However, the oasis consensus type for block height is `int64`, so the indexer `consensus` types thus use `int64` as well. Eth's ethclient [block number](https://pkg.go.dev/github.com/ethereum/go-ethereum@v1.10.25/core/types#Header) is `big.Int`, so I'm wondering if it'd be better for us to use `big.Int` instead?

Reply:
1. Go's conversions from uint64 to int64 is silent about stuff going out of range, and we don't want to write our own checked conversions if we don't have to.
2. The Oasis runtime SDK uses uint64 for round, so there has to be some uint64 somewhere in the system.
3. The database is using bigint (equivalent to int64) for round, so there has to be some int64 somewhere in the system.
4. The database errors out if a value is out of range.

so the minimal solution to all this is for the Go code to use uint64 on the analysis side and let the database detect out of range errors

Reply:
Got it, thanks for the detailed explanation!

Reply:
> Go's conversions from uint64 to int64 is silent about stuff going out of range

Already got bit by this.

Reply:
what happened

Reply:
Technically underflow fixed in #145. I had an epoch of `(0 - 1)` being too big.